docs/source/reference/core/configuration.rst

```diff
@@ -0,0 +1,102 @@
+.. _configuration:
+
+Configuring Chainer
+===================
+
+..  currentmodule:: chainer
+
+Chainer provides some global settings that affect the behavior of some functionalities.
+Such settings can be configured using the *unified configuration system*.
+The system provides a transparent way to manage the configuration for each process and for each thread.
+
+The configuration is managed by two global objects: :data:`chainer.global_config` and :data:`chainer.config`.
+The :data:`global_config` object maintains the configuration shared in the Python process.
+This is an instance of the :class:`~chainer.configuration.GlobalConfig` class.
+It can be used just as a plain object, and users can freely set any attributes to it.
+The :data:`config` object, on the other hand, maintains the configuration for the current thread.
+It behaves like a thread-local object, and any attribute modifications are only visible to the current thread.
+If no value is set to this thread-local configuration for a given key, the global configuration is transparently referred.
+Thanks to this transparent lookup, users can always use the ``config`` object to read any configuration so that the thread-local configuration is used if available and otherwise the default global setting is used.
+The ``config`` object is an instance of the :class:`~chainer.configuration.LocalConfig` class.
+
+The following entries of the configuration are currently provided by Chainer.
+Some entries support environment variables to set the default values.
+Note that the default values are set to the global config.
+
+``chainer.config.debug``
+   Debug mode flag.
+   If it is ``True``, Chainer runs in the debug mode.
+   See :ref:`debug` for more information of the debug mode.
+   The default value is given by ``CHAINER_DEBUG`` environment variable (set to 0 or 1) if available, otherwise uses ``False``.
+``chainer.config.enable_backprop``
+   Flag to enable backpropagation support.
+   If it is ``True``, the default behavior of :class:`Function` application to :class:`Variable` is non-volatile if all inputs has ``AUTO`` volatile flag.
+   Otherwise, the default behavior is set to volatile mode.
```

Comment:
How about adding an explanation about "volatile mode"? For example,

~is set to volatile mode that doesn't keep track of any function applications in all :class:`Variable`s.